content/en/blog/_posts/2023-12-20-disabling-in-tree-cloud-provider-goes-beta.md

```diff
@@ -0,0 +1,200 @@
+```
+
+_Please note that if you have other feature gate modifications on the command
+line, they will need to include these 2 feature gates._
+
+### Upgrading on other providers
+
+For providers other than Azure, GCE, or vSphere, good news, the external cloud
+controller manager should be already be in use. The code for AWS and OpenStack
+providers has been removed from Kubernetes before version 1.27 was released.
+Other providers beyond the 5 listed in this post were never included in
+Kubernetes and as such they began their life as external cloud controller
+managers.
+
+### Upgrading from older Kubernetes versions
+
+If you are upgrading from a Kubernetes release older than 1.26, and you are on
+AWS, Azure, GCE, OpenStack, or vSphere then you will need to enable the
+`--cloud-provider=external` flag, and follow the advice for installing and
+running a cloud controller manager for your provider.
+
+Please read the documentation for
+[Cloud Controller Manager Administration][ccmadmin] and
+[Migrate Replicated Control Plane To Use Cloud Controller Manager][ccmha]. See
+below for links to specific cloud provider implementations.
+
+## Where to find a cloud controller manager?
+
+At its core, this announcement is about the cloud provider integrations that
+were previously included in Kubernetes. As these components move out of the
+core Kubernetes code and into their own repositories, it is important to note
+a few things:
+
+First, SIG Cloud Provider maintains a reference framework for developers who
+wish to create cloud controller managers for any provider. See the
+[cloud-provider repository][cloud-provider] for more information about how
+these controllers work and how to get started creating your own.
+
+Second, there are many cloud controller managers available for Kubernetes.
+This post is addressing the providers that have been historically included
+with Kubernetes but are now in the process of being removed. If you need a
+cloud controller manager for your provider and do not see it listed here,
+please reach out to the cloud provider you are integrating with or the
+[Kubernetes SIG Cloud Provider community][sig] for help and advice. It is
+worth noting that while most cloud controller managers are open source today,
+this may not always be the case. Users should always contact their cloud
+provider to learn if there are preferred solutions to utilize on their
+infrastructure.
+
+### Cloud provider integrations provided by the Kubernetes project {#cloud-provider-integrations}
+
+* AWS - https://github.com/kubernetes/cloud-provider-aws
+* Azure - https://github.com/kubernetes-sigs/cloud-provider-azure
+* GCE - https://github.com/kubernetes/cloud-provider-gcp
+* OpenStack - https://github.com/kubernetes/cloud-provider-openstack
+* vSphere - https://github.com/kubernetes/cloud-provider-vsphere
+
+## Want to learn more?
+
+Cloud providers and cloud controller managers serve a core function in
```

Comment:
```suggestion
Cloud provider integrations, and the core library for cloud controller managers, serve a core function in
```
?

Reply:
the cloud-provider framework isn't required per se, i was trying to call out the CCMs themselves here. like both the provider (the actual infrastructure) and the ccm are core to this functionality. maybe that was implying too much?